File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) 
    {
      //just count original when search pointer's value is not eqaul to original's
        int original = 0;

        for(int search = 0; search < nums.Length; search++){
          if(original < 2 || nums[search] != nums[original -2]){
            nums[original] = nums[search];
            original++;

          }

        }
        return  original;

    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
      #idea : [0,0,0,1,1]-> [0,0,1,1,-]
      # [0,0]
      k = 0
      for i in range(len(nums)):
        if k < 2 or nums[i] != nums[k - 2]:
          nums[k] = nums[i]
          k+=1
      return k


        